### ISSUE TITLE
TODO before deployment of Bonding V2 - collect a list of all Bonding V1 users' deposits

### ISSUE THREAD

**Issue by zgorizzo69** [ubiquity/ubiquity-dollar#156]
### Make a list for people that should migrate

- For every address that successfully deposited 
create a **deposit list** to record  the address, LP Amount and weeks 

- For every address that successfully withdraw
go through the **deposit list** 

If there is only one deposit for that address remove the record from the deposit list as the user as already withdrawn.

If there is more than one deposit we have to find out which one of the deposit made by that address corresponds to that withdraw. to do so we can check inside the logs the `TransferSingle` events of the deposit to get the ID of the bondingshare minted during the deposit and compare it to the withdraw id argument . Once we are sure that we get the right deposit corresponding to the withdraw remove it from the **deposit list** 
ex with this deposit [ tx](https://etherscan.io/tx/0x117129bd2e77a3774b8ff8e52323b9c6909d3ec278cdb0e78446537619e49d49#eventlog) the id is : 12827000
![image](https://user-images.githubusercontent.com/16853695/127030548-984bd24f-4c54-40b6-ad55-88cc47b873ae.png)
and the corresponding withdraw is [here](https://etherscan.io/tx/0x568616b226b2bd0d493bc30b512dd8f53545172e1029934a696c7f6a7855c26c) : 
![image](https://user-images.githubusercontent.com/16853695/127030646-a6e5276b-e884-45f4-ae1e-505a4e2e3a0a.png)

The **deposit list** should then be separated into three arrays that I will provide as an argument of the bondingV2 constructor:
One array for the address , One for the LP amout in wei One for the number of weeks . It is utterly important that the array is ORDERED meaning that address[0], LP[0], weeks[0] corresponds to one record of the deposit list. So the arrays should be the same length. It should look like this 
```javascript
 const toMigrateOriginals = [
    "0x89eae71b865a2a39cba62060ab1b40bbffae5b0d",
    "0x10693e86f2e7151b3010469e33b6c1c2da8887d6",
    "0xa53a6fe2d8ad977ad926c485343ba39f32d3a3f6",
    "0x7c76f4db70b7e2177de10de3e2f668dadcd11108",
    "0x4007ce2083c7f3e18097aeb3a39bb8ec149a341d",
    "0x0709b103d46d71458a71e5d81230dd688809a53d",
    "0xf6501068a54f3eab46c1f145cb9d3fb91658b220",
    "0xd028babbdc15949aaa35587f95f9e96c7d49417d",
    "0xa1c7bd2e48f7f3922e201705f3491c841135f483",
    "0x9968efe1424d802e1f79fd8af8da67b0f08c814d",
  ];
  const toMigrateLpBalances = [
    "44739174270101943975392",
    "74603879373206473097231",
    "618000000000000000000",
    "374850000000000000000",
    "1878674425540571814543",
    "44739174270101943975392",
    "74603879373206473097231",
    "618000000000000000000",
    "374850000000000000000",
    "1878674425540571814543",
  ];
  const toMigrateWeeks = [
    "208",
    "100",
    "100",
    "11",
    "1",
    "100",
    "208",
    "208",
    "208",
    "4",
  ];
```

**Comment by Zequez** [comment]
This is to be created as a library? As a hardhat task?

**Comment by zgorizzo69** [comment]
as you want as long as it is fast ,you can use excel for instance, I am only interested in the end result and this should be done only once

**Comment by zgorizzo69** [comment]
If there is several deposit for one address they should be merged so we sum the LP amount and for the duration we do a weighted average instead of taking the   period sum(LP1*Duration1.....LPn*Durationn)/sum(LP1...LPn)
ex if 0x42 deposited twice 0.1LP one for 1 week and a second one for 208 weeks we should only have one
record in the deposit list 0x42 0.2LP and (0,1*1+0,1*208)/(01+0,1)=104.5  weeks

**Comment by Zequez** [comment]
Something came up: what would happen if there is a partial withdrawal, when the function is called with a sharesAmount that is less than the total shares for that bondingShare ID?

**Comment by Zequez** [comment]
Current implementation doesn't consider the mentioned situation; currently implementing solution.

**Comment by zgorizzo69** [comment]
fyi if you do ` .setAction(async (taskArgs: CliArgs, { ethers }) => {`
you can then use ethers and then ditch the import of both ethers, web3 and ABI 
because you can then do 
` const metaPool = (await ethers.getContractAt(
        "IMetaPool",
        "0x20955CB69Ae1515962177D164dfC9522feef567E"
      )) as IMetaPool;` to get a typed contract (and its ABI but as it is typed you don't need it anymore)
plus access all the helpers funcs

**Comment by Zequez** [comment]
Oh cool that seems like the better approach, I'll test it out. I'll also see how to remove the web3 dependency and use ethers instead.

**Comment by zgorizzo69** [comment]
merged in ubiquity/uad-contracts#25
